app/models/user.rb

```diff
@@ -412,6 +412,10 @@ def self.find_by_username(username)
     find_by(username_lower: normalize_username(username))
   end
 
+  def in_any_groups?(group_ids)
```

Comment:
@tgxworld the second change, checks if the user is in any of the passed in group IDs based on their `GroupUser` relation via:

https://github.com/discourse/discourse/blob/15119b524d13f9842836aa4dcc53672569b3283c/app/models/user.rb#L87-L87

Usage with site setting change:

```ruby
@user.in_any_groups?(SiteSetting.group_setting_map(:personal_message_enabled_groups))
```

I guess the main improvement we could do here is a stripped down scope that just gets the group IDs and names?